tests/test_integration.py

```diff
@@ -144,6 +145,12 @@ def test_to_tree(self):
         tree = parse_tree(data)
         self.assertEqual([toklist.to_tree() for toklist in toklists], tree)
 
+    def test_parse_incr(self):
+        self.assertEqual(parse(data), list(parse_incr(StringIO(data))))
+
+    def test_parse_tree_incr(self):
+        self.assertEqual(parse_tree(data), list(parse_tree_incr(StringIO(data))))
+
```

Comment:
Isn't BUF_SIZE larger than all of data? I might be wrong. That would mean this tests doesn't really test reading the data in chunks, just how it behaves on the first chunk.